osidb/models.py

```diff
@@ -1350,7 +1352,8 @@ def _perform_bzsync(self, *args, bz_api_key, **kwargs):
 
         # sync to Bugzilla
         bs = FlawBugzillaSaver(self, bz_api_key)  # prepare data for save to BZ
-        self = bs.save()  # actually send to BZ (but not save to DB)
+        old_instance = kwargs.pop("old_instance", None)
+        self = bs.save(old_instance)  # actually send to BZ (but not save to DB)
```

Comment:
What is `self` now?

Technically, `self` is one of the parameters. It's a variable that exists in this scope only. Assigning anything to it just changes the value. The original object referenced by the previous reference saved in the variable named `self` is not changed.

Before this PR, this weird construction conveyed (or so I hoped, since it was there for quite some time) that `bs.save()` actually returns the identical reference to the object referenced by `self` that was passed to `FlawBugzillaSaver(self, ...`.

Assigning something else than the original reference to the variable named `self` is technically valid in Python, but semantically it's confusing. If we do that, we can also create variables that shadow built-in functions. 

While I didn't find a PEP that would directly discourage this, there's https://peps.python.org/pep-0008/#function-and-method-arguments that says _"Always use `self` for the first argument to instance methods."_.

This is a slippery slope :) We don't want ending up with writing programs that have something like this in it :)

```
int = 1
id = 2
str = 3
bool = 4
len = 5
all = 6
list = 7
dict = 8
print = 9
type = 10
open = 11
```

So...

```suggestion
        flaw_instance = bs.save(old_instance)  # actually send to BZ (but not save to DB)
```
... and then later use that for the rest of the calls that use `self`

Reply:
Agreed, assigning to self is very confusing

Reply:
Thank you!